### ISSUE TITLE
Explain manual and automated key backup better

### ISSUE THREAD

**Issue by aaronraimist** [vector-im/element-meta#1886]
Lots of people tend to think manual key backup is something like GPG, ie if you export the key as soon as you create your account you'll be able to decrypt all future messages that are sent after you exported the keys.

**Comment by 35609902357** [comment]
>I don't think that is an option?

It is not. I opened vector-im/element-web#13263.

>Also for the other two options, the recovery key is not being backed up AFAIK, the keys for decrypting messages are. It is always on you to store the recovery key.

The Recovery key is actually backed up. Read @cyphar's [good explanation](https://github.com/vector-im/riot-web/issues/8751#issuecomment-484162224) about it.

**Comment by aaronraimist** [comment]
OK yes I misread your wording. Yes the recovery key can be backed up in an encrypted format but that doesn't do anything for you if you lose the password.

I don't think vector-im/element-web#13263 is a good idea. Key backup needs to get simpler (less options), not more complex (more options).

Imo the UI should be something like this:

----
Your private chats are end-to-end encryption by default (encryption can be enabled in other rooms in the room's settings). This means that messages can be read only by you and your recipient(s) and no one else, not even server administrators.

To ensure you don't lose message history you can backup the message keys. Riot can automatically back up message keys up in an encrypted format to your homeserver or you can manually download them to your computer.

If you chose to do manual key backups, they must be done regularly to ensure you don't lose messages since encryption keys change constantly to ensure Perfect Forward Secrecy. The downloaded key file can only decrypt messages sent before you pressed the download keys button. It cannot decrypt any future keys. Automatic key backup is recommended for this reason.

[ Button: Use automatic key backup ]     [ Button: Use manual key backup ]

----

If the user chooses to use manual key backup then all of the automated key backup stuff is hidden and can be shown again by some small button saying switch to automatic key backup.

If the user chooses to use automatic key backup then the user is walked through setting that up. Imo the only way to set this up should be to use a recovery key which the user is told to store in a password manager or print and store securely. Having a second password is not great because most users are going to get confused on what the difference is between the two passwords and will likely either use the same password for both or will use a less secure password.

The manual key backup buttons are still available in this mode but are smaller/de-emphasized.

**Comment by 35609902357** [comment]
@aaronraimist 
Word it like that and you are guaranteed to push away 100% of users who you will propose Riot to. vector-im/element-web#13263 will grant non-technical users a familiar experience. For them step 2 is one step too much. If you tell them "download this app, but be sure you regularly backup keys and store the Recovery key in a safe place, else you may loose chat history" and you will make sure they continue using Zoom forever. How many people need manual backup? I would go as far as to say vector-im/element-web#13263 should be the default and the other options should be unlockable clicking a toggle, all in order to ensure ease of use by non-techhies, but I think the multiple choice I wrote should be a happy medium for everyone. GPG failed because it was not easy to use, for non-techies to use a software it must be idiot-proof, otherwise enjoy your loneliness in Matrix while the others hang out in Zoom, which you yourself will be forced to use some way or another. I wrote it like that because I don't want to explain users how they should use the application, I want to say "Install Riot and let's hang out there" and let them understand everything by themselves on first use.

**Comment by aaronraimist** [comment]
Related:  vector-im/element-meta#1474

**Comment by c33s** [comment]
![nag screen](https://user-images.githubusercontent.com/649209/82042817-d1ff5600-96aa-11ea-9e37-3caac5f44653.png)

well i felt really forced to use cloud backup with this nag screen comming up every time i started riot (where i asked for clear description in the app vector-im/element-web#13685). 
i would love to have a local auto-backup which saves me from loosing access to my message history (*) because after my harddisk ran full i simply lost my keys/access to all my message history vector-im/element-web#15788 (i assume that it happend because of key rotation but why keys are getting rotated if no space is left on the device? (freeing up space and restarting the app has not brought back my messages/keys))

*: i really miss a feature to permanent decrypt a message locally. i someone is on my system where they keys reside the encrypted db wont really help

**Comment by brumasribera** [comment]
Would there be a way to simply back up the conversations and get them when you log in into new devices. I am really not concerned about security and this huge complexity I find. 

I only would like to be able to keep the whole history of messages all the time, and this seems to be impossible right now to be done on a human way, not having to export files with encrypted keys now and then.

**Comment by aaronraimist** [comment]
@brumasribera Yes. That is what the Secure Backup feature does. It automatically backs up your keys. As long as you provide your Security Key (or Security Phrase) each time you login, your keys will be automatically backed up and restored.

**Comment by brumasribera** [comment]
@aaronraimist Backing up key means also backing up messages, I am more concerned with messages and not so much on those keys. 

For me, it would be just more simple to recover the messages normally once reconnected to your account and have this extra safety layer optional.

Or at least be able to disable it, as I find it a bit cumbersome for my normal use. I am of course only talking about UX experience for the user. Thank you for your time!

**Comment by aaronraimist** [comment]
@brumasribera 

> Backing up key means also backing up messages, I am more concerned with messages and not so much on those keys.

Well that's a different problem. If you don't trust your homeserver to not delete your messages then you probably want to be using the export chat feature. Go to the Room info page and then click Export chat.  There is no feature to automatically do that though, that would be https://github.com/vector-im/element-meta/issues/1388.

> For me, it would be just more simple to recover the messages normally once reconnected to your account and have this extra safety layer optional.

Encryption makes it very difficult to automatically restore messages. If you want messages to be automatically visible with no action other than logging in, your only real option today is to not encrypt the room. In the future https://github.com/matrix-org/matrix-spec-proposals/pull/3265, https://github.com/matrix-org/matrix-spec-proposals/pull/2957, or https://github.com/matrix-org/matrix-spec-proposals/pull/3262 may solve this.
